server/pbench/bin/index-pbench

```diff
@@ -33,8 +33,10 @@ _DEBUG    = 0
 # global - defaults to normal dict, ordered dict for unittests
 _dict_const = dict
 
-# 30 second timeout talking to Elasticsearch
-_read_timeout = 30
+# 100,000 minute timeout talking to Elasticsearch; basically we just don't
+# want to timeout waiting for Elasticsearch and then have to retry, as that
+# can add undue burden to the Elasticsearch cluster.
+_read_timeout = 100000*60
```

Comment:
I find it a bit strange that _read_timeout is defined in index-pbench, but _request_timeout is in the pbench module. I would expect both of them to be defined in the module and not be controlled by the client. Do you have plans to do that further down the road?

Reply:
Because `es_index` is using the `_request_timeout` to opine that bulk requests should not timeout, no matter what the Elasticsearch client object, `es`, has for a read timeout.

The caller of `es_index` is responsible for setting up the Elasticsearch connection client object, which is where the `_read_timeout` is applied.

So `_read_timeout` lives in `index-pbench` because it sets up the Elasticsearch connection client object, and `_request_timeout` lives in `es_index` because it wants to make sure bulk requests don't timeout early.